apis/rust/node/src/lib.rs

```diff
@@ -140,15 +153,15 @@ impl Drop for DoraNode {
 }
 
 #[derive(Debug)]
-pub struct Input {
-    pub id: DataId,
+pub struct Message {
     pub data: Vec<u8>,
-    pub input_context: DoraInputContext,
+    pub metadata: Metadata,
 }
 
-#[derive(Debug)]
-pub struct DoraInputContext {
-    pub open_telementry: String,
+#[derive(Debug, Clone)]
+pub struct Metadata {
+    pub id: DataId,
+    pub otel_context: String,
 }
```

Comment:
What was the motivation for moving `id` from `Input` to `Metadata`? I think it's better to keep it next to `data` because it behaves different than other metadata. It is dependent on the mapping in the dataflow YAML file and a message's output ID can be different from the input IDs under which it is received by other nodes (because of input renames).

Reply:
I agree that `id` is slightly different from the other metadata as it might change depending on the mapping. We can choose to keep `id` to explicitly display this distinction. I just felt like it was in a similar bag of information that's why I grouped them.

Reply:
To me, `metadata` feels like something that was actually sent over the network, similar to data. So filling in additional, receiver-specific data upon reception feels like a potential source of confusion. So I would prefer to keep it separate.

Reply:
Ok, sure, make sense.